### PR TITLE
perception/simulation/desktop-full tags to new ros2 images

### PR DESCRIPTION
**Pull request by mikaelarguedas**
Superseed https://github.com/osrf/docker_images/pull/619

keeping this as draft waiting for answers to https://github.com/osrf/docker_images/pull/620#discussion_r870638478 and https://github.com/osrf/docker_images/pull/619#issuecomment-1124132514 and make changes accordingly before placing in revier

### CODE REVIEW

File: ros/.config/ros2/Makefile.em [hunk 1]
```diff
@@ -39,3 +42,17 @@ ci_buildx:
 		docker push \
 			osrf/ros:$release_name-desktop; \
 	fi
+	@docker buildx build --pull --push \
```
Comment (mikaelarguedas) [inline ros/.config/ros2/Makefile.em:45]: @ruffsl I'm not familiar with this part. is this the right way to do this ?

Comment (ruffsl) [reply ros/.config/ros2/Makefile.em:45]: This looks fine 👍

File: ros/rolling/ubuntu/jammy/images.yaml.em [hunk 2]
```diff
@@ -28,3 +28,11 @@ images:
...
         ros2_packages:
             - desktop
+    desktop-full:
+        base_image: osrf/@(user_name):@(ros2distro_name)-desktop-@(os_code_name)
+        maintainer_name: @(maintainer_name)
+        template_name: docker_images_ros2/create_ros_image.Dockerfile.em
+        template_packages:
+            - docker_templates
+        ros2_packages:
+            - desktop
```
Comment (ruffsl) [inline ros/rolling/ubuntu/jammy/images.yaml.em:38]: Shouldn't this be `desktop-full`?

File: ros/rolling/ubuntu/jammy/desktop-full/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# This is an auto generated Dockerfile for ros:desktop-full
+# generated from docker_images_ros2/create_ros_image.Dockerfile.em
+FROM osrf/ros:rolling-desktop-jammy
+
+# install ros2 packages
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    ros-rolling-desktop=0.9.3-2* \
```
Comment (ruffsl) [inline ros/rolling/ubuntu/jammy/desktop-full/Dockerfile:7]: Shouldn't this be `desktop-full`? Check template args.

**Review by ruffsl — CHANGES_REQUESTED** [review]

**Comment by mikaelarguedas** [comment]
Updated to add simulation and perception

@tfoote could you or the person in charge of Humble ping here once these new variants have been synced to main? this is a prerequisite for the dockerfiles to be generated and not break CI by merging this

**Comment by tfoote** [comment]
I'll try to remember after the next syncs but also pinging release managers @audrow and @nuclearsandwich

**Comment by audrow** [comment]
@mikaelarguedas, is this PR waiting on anything before it can come out of draft status? It'd be nice to get this in just after the Humble sync on Monday the 23rd.

**Comment by mikaelarguedas** [comment]
It is waiting for a humble (and a rolling) sync to main. Those first syncs ideally happen during the last week of the beta phase of the distro. This gives time for the images to be submitted upstream and have time to build on all architectures. For example if the sync happens on the 23rd, the images will be available around only on the 26th.

Do you know when the rolling and humble sync will happen?

**Comment by mikaelarguedas** [comment]
Waiting for the new variants to be unblocked by the rolling sync, a minimal PR adding humble images has been made separately at https://github.com/osrf/docker_images/pull/622 and submitted to the official docker images repos

**Comment by audrow** [comment]
@mikaelarguedas, the Humble sync has just finished. The rolling sync should be this week.

Also, could we have a smaller PR to enable Docker images up to desktop on Humble before we add the perception/simulation/desktop-full tags? I think it would be good to reduce the number of moving parts.

**Comment by audrow** [comment]
> Do you know when the rolling and humble sync will happen?

The Humble sync happened today, the next rolling sync should happen this week.

**Comment by nuclearsandwich** [comment]
https://discourse.ros.org/t/preparing-for-rolling-sync-2022-05-26/25731

**Comment by mikaelarguedas** [comment]
Upstream PR: https://github.com/docker-library/official-images/pull/12521

**Pull request merged**